torchtune/utils/_checkpointing/_checkpointer_utils.py

```diff
@@ -54,9 +57,16 @@ def safe_torch_load(checkpoint_path: Path) -> Dict[str, Any]:
     try:
         # convert the path into a string since pathlib Path and mmap don't work
         # well together
-        state_dict = torch.load(
-            str(checkpoint_path), map_location="cpu", mmap=True, weights_only=True
-        )
+        if str(checkpoint_path).endswith(".safetensors") and is_safetensors_available():
```

Comment:
Not a fan of this approach. Can we just add a key to the config, something like ```is_safetensors_file``` and then based on the value determine if we use torch.load or not. Also please break this down into a sub function (eg: ```load_from_safetensor``` or something similar.

@joecummings WDYT?